python_ta/typecheck/base.py

```diff
@@ -52,33 +99,70 @@ class TypeFail(TypeResult):
     """Represents the result of a failed type check operation
     Contains error message
     """
-
-    def __init__(self, tnode1: _TNode = None, tnode2: _TNode = None, src_node: NodeNG = None):
-        if isinstance(tnode1, str):
-            self.msg = tnode1
-            self.tnode1 = None
-        else:
-            self.tnode1 = tnode1
-            self.msg = ""
-        self.tnode2 = tnode2
-        self.src_node = src_node
+    def __init__(self, msg: str):
```

Comment:
Make this attribute `Optional[str]`. It's not necessary. In the subclass below you're overriding `__str__`, which is good, but this means that `msg` isn't useful there.